### ISSUE TITLE
cURL will not use TLS1.2 unless forced. CentOS6.  Tested with PayPal

### ISSUE THREAD

**Issue by MikeParkin** [curl/curl#700]
### PayPal TLS1.2 Update
PayPal require TLS1.2 after June 17th 2016: [Read More](https://www.paypal-knowledge.com/infocenter/index?page=content&widgetview=true&id=FAQ1914&viewlocale=en_US)

My curl/libcurl will not connect to their testing address, unless I force TLS1.2. This is causing me difficulty as it also affects PHP (same behaviour), and updating all the code to force TLS1.2 is not really workable.

### Issue
Using curl without forcing tls version:
```
$ curl https://tlstest.paypal.com
curl: (35) SSL connect error
```
Forcing TLS1.2:
```
$ curl --tlsv1.2 https://tlstest.paypal.com/
PayPal_Connection_OK
```
### PHP Behaviour
PHP shows the same behaviour as curl:
```
$ php -r '$ch = curl_init(); curl_setopt($ch, CURLOPT_URL, "https://tlstest.paypal.com/"); var_dump(curl_exec($ch)); var_dump(curl_error($ch));'
bool(false)
string(17) "SSL connect error"
```
Again, forcing 1.2 works:
```
$ php -r '$ch = curl_init(); curl_setopt($ch, CURLOPT_SSLVERSION, 6); curl_setopt($ch, CURLOPT_URL, "https://tlstest.paypal.com/"); var_dump(curl_exec($ch)); var_dump(curl_error($ch));'
PayPal_Connection_OKbool(true)
string(0) ""
```

### curl/libcurl version
```
curl 7.19.7 (x86_64-redhat-linux-gnu) libcurl/7.19.7 NSS/3.19.1 Basic ECC zlib/1.2.3 libidn/1.18 libssh2/1.4.2
Protocols: tftp ftp telnet dict ldap ldaps http file https ftps scp sftp 
Features: GSS-Negotiate IDN IPv6 Largefile NTLM SSL libz 
```
```
curl.x86_64                        7.19.7-46.el6                      @base   
libcurl.x86_64                     7.19.7-46.el6                      @base     
python-pycurl.x86_64               7.19.0-8.el6                       @base   
```

### operating system
```
$ cat /etc/*release
CentOS release 6.5 (Final)
LSB_VERSION=base-4.0-amd64:base-4.0-noarch:core-4.0-amd64:core-4.0-noarch
CentOS release 6.5 (Final)
CentOS release 6.5 (Final)
```

**Comment by kdudka** [comment]
You are reporting the problem at a wrong place!  Anyway, this is going to be fixed in RHEL-6.8:

https://bugzilla.redhat.com/1272504
https://bugzilla.redhat.com/1289205

**Comment by MikeParkin** [comment]
Thank you, much appreciated.
